drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/GitHubSource.java

```diff
@@ -20,120 +28,133 @@
  */
 public abstract class GitHubSource implements ExternalBinarySource {
 
-    private Logger log = Logger.getLogger(GitHubSource.class.toString());
-
-    private String projectUrl;
+    private static final String LATEST_URL = "/releases/latest";
+    private static final String RELEASES_URL = "/releases";
 
-    // URL suffixes
-    private String latestUrl = "/releases/latest";
-    private String releasesUrl = "/releases";
+    private static final Logger log = Logger.getLogger(GitHubSource.class.toString());
 
     // JSON keys
     private String tagNameKey = "tag_name";
-    private String idKey = "id";
     private String assetNameKey = "name";
     private String browserDownloadUrlKey = "browser_download_url";
     private String assetsKey = "assets";
 
-    private ExternalBinary binaryRelease;
+    private final HttpClient httpClient;
+    private final GitHubLastUpdateCache cache;
+    private final Gson gson;
+    private final String projectUrl;
+    private final String uniqueKey;
 
     /**
      * @param organization GitHub organization/user name the project belongs to
-     * @param project GitHub project name
+     * @param project      GitHub project name
      */
-    public GitHubSource(String organization, String project) {
-        projectUrl = String.format("https://api.github.com/repos/%s/%s", organization, project);
+    public GitHubSource(String organization, String project, HttpClient httpClient, GitHubLastUpdateCache gitHubLastUpdateCache) {
+        this.httpClient = httpClient;
+        this.projectUrl = String.format("https://api.github.com/repos/%s/%s", organization, project);
+        this.uniqueKey = organization + "@" + project;
+        this.gson = new Gson(); // TODO think if that should be really a field
```

Comment:
It's thread safe, so we can make it static, but I'm not sure if we benefit something from it. It doesn't maintain any state between invocations though.